pkg/vm/engine/tpe/tuplecodec/indexhandler.go

```diff
@@ -318,11 +324,231 @@ func (ihi * IndexHandlerImpl) WriteIntoIndex(writeCtx interface{}, bat *batch.Ba
 	return nil
 }
 
-func (ihi * IndexHandlerImpl) DeleteFromTable(table *descriptor.RelationDesc, bat *batch.Batch) error {
-	panic("implement me")
+func (ihi * IndexHandlerImpl) DeleteFromTable(writeCtx interface{}, bat *batch.Batch) error {
+	return ihi.DeleteFromIndex(writeCtx, bat)
 }
 
-func (ihi * IndexHandlerImpl) DeleteFromIndex(index *descriptor.IndexDesc, attrs []descriptor.AttributeDesc, bat *batch.Batch) error {
-	panic("implement me")
+func (ihi * IndexHandlerImpl) DeleteFromIndex(writeCtx interface{}, bat *batch.Batch) error {
+	indexWriteCtx, ok := writeCtx.(*WriteContext)
+	if !ok {
+		return errorWriteContextIsInvalid
+	}
+
+	if bat == nil {
+		return nil
+	}
+	//1.encode prefix (tenantID,dbID,tableID,indexID)
+	tke := ihi.tch.GetEncoder()
+	var prefix TupleKey
+	prefix,_ = tke.EncodeIndexPrefix(prefix,
+		uint64(indexWriteCtx.DbDesc.ID),
+		uint64(indexWriteCtx.TableDesc.ID),
+		uint64(indexWriteCtx.IndexDesc.ID))
+
+	indexWriteCtx.callback = callbackPackage{
+		prefix: prefix,
+	}
+
+	// get every row of the delete set
```

Comment:
The logic here is almost the same as that in the 'func (ba *BatchAdapter) ForEachTuple'. Try to merge the logic into the 'func (ba *BatchAdapter) ForEachTuple(' instead of the copy another one.